crates/worker/src/commands/compile.rs

```diff
@@ -0,0 +1,297 @@
+};
+
+pub struct CompilationFile {
+    // legacy name. file_path really
+    pub file_name: String,
+    pub file_content: Vec<u8>,
+}
+
+pub struct CompilationInput {
+    pub config: CompilationConfig,
+    // legacy. files really
+    pub contracts: Vec<CompilationFile>,
+}
+
+pub struct CompilationArtifact {
+    pub file_name: String,
+    pub file_content: Vec<u8>,
+    pub is_contract: bool,
+}
+
+pub async fn compile(
+    request: CompilationRequest,
+    db_client: &DynamoDBClient,
+    s3_client: &S3Client,
+) -> Result<(), CompilationError> {
+    let item = db_client.get_item(request.id.clone()).await?;
+    let item: Item = match item {
+        Some(item) => item,
+        None => {
+            error!("No item id: {}", request.id);
+            return Err(CompilationError::NoDBItemError(request.id));
+        }
+    };
+
+    match item.status {
+        Status::Pending => {}
+        status => {
+            warn!("Item already processing: {}", status);
+            return Err(CompilationError::UnexpectedStatusError(status.to_string()));
+        }
+    }
+
+    let dir = format!("{}/", request.id);
+    let files = s3_client.extract_files(&dir).await?;
+    {
+        let db_update_result = db_client
+            .client
+            .update_item()
+            .table_name(db_client.table_name.clone())
+            .key("ID", AttributeValue::S(request.id.clone()))
+            .update_expression("SET #status = :newStatus")
+            .condition_expression("#status = :currentStatus")
+            .expression_attribute_names("#status", "Status")
+            .expression_attribute_values(
+                ":newStatus",
+                AttributeValue::N(u32::from(Status::Compiling).to_string()),
+            )
+            .expression_attribute_values(
+                ":currentStatus",
+                AttributeValue::N(u32::from(Status::Pending).to_string()),
+            )
+            .send()
+            .await;
+        match db_update_result {
+            Ok(_) => {}
+            Err(SdkError::ServiceError(err)) => match err.err() {
+                UpdateItemError::ConditionalCheckFailedException(_) => {
+                    error!("Conditional check not met");
+                    return Err(CompilationError::UnexpectedStatusError(
+                        "Concurrent status change from another instance".into(),
+                    ));
+                }
+                _ => return Err(DBError::from(SdkError::ServiceError(err)).into()),
+            },
+            Err(err) => return Err(DBError::from(err).into()),
+        }
+    }
+
+    match do_compile(
+        request.id.clone(),
+        CompilationInput {
+            config: request.config,
+            contracts: files,
+        },
+    )
+    .await
+    {
+        Ok(val) => Ok(on_compilation_success(&request.id, db_client, s3_client, val).await?),
+        Err(err) => match err {
```

Comment:
subjective suggestion: you can unwrap this match into the outer one, so it will look like

```suggestion
Err(CompilationError::CompilationFailureError(value)) => ...
```

Reply:
Yeap, that is prettier. Done